specs/filesystem_hierarchy.md

```diff
@@ -0,0 +1,529 @@
+application finds `$XDG_CACHE_HOME` set, it should use the directory specified
+in it instead of this directory.
+
+### `~/.config/`
+
+Application configuration. When a new user is created, this directory will be
+empty or not exist at all. Applications should fall back to defaults should
+their configuration in this directory be missing. If an application finds
+`$XDG_CONFIG_HOME` set, it should use the directory specified in it instead of
+this directory.
+
+### `~/.local/bin/`
+
+Executables that shall appear in the user's `$PATH` search path. It is
+recommended not to place executables in this directory that are not useful for
+invocation from a shell; these should be placed in a subdirectory of
+`~/.local/lib/` instead. Care should be taken when placing
+architecture-dependent binaries in this place, which might be problematic if the
+home directory is shared between multiple hosts with different architectures.
+
+### `~/.local/lib/`
+
+Static, private vendor data that is compatible with all architectures.
+
+### `~/.local/lib/arch-id/`
+
+Location for placing public dynamic libraries. The architecture identifier to
+use is defined on [Multiarch Architecture Specifiers
+(Tuples)](https://wiki.debian.org/Multiarch/Tuples) list.
+
+### `~/.local/share/`
+
+Resources shared between multiple packages, such as fonts or artwork. Usually,
+the precise location and format of files stored below this directory is subject
+to specifications that ensure interoperability. If an application finds
+`$XDG_DATA_HOME` set, it should use the directory specified in it instead of
+this directory.
+
+### `~/.local/state/`
+
+Application state. When a new user is created, this directory will be empty or
+not exist at all. Applications should fall back to defaults should their state
+in this directory be missing. If an application finds `$XDG_STATE_HOME` set, it
+should use the directory specified in it instead of this directory.
+
+## Write Access
+
+### Unprivileged Write Access
+
+Unprivileged processes generally lack write access to most of the hierarchy.
+
+The exceptions for normal users are `/tmp/`, `/var/tmp/`, `/dev/shm/`, as well
+as the home directory `$HOME` (usually found below `/home/`) and the runtime
+directory `$XDG_RUNTIME_DIR` (found below `/run/user/`) of the user, which are
+all writable.
+
+For unprivileged system processes, only `/tmp/`, `/var/tmp/` and `/dev/shm/` are
+writable. If an unprivileged system process needs a private writable directory
+in `/var/` or `/run/`, it is recommended to either create it before dropping
+privileges in the daemon code, to create it via
+[`tmpfiles.d(5)`](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html)
+fragments during boot, or via the `StateDirectory=` and `RuntimeDirectory=`
+directives of service units (see
+[`systemd.unit(5)`](https://www.freedesktop.org/software/systemd/man/systemd.unit.html)
+for details).
```

Comment:
I think we'd want to change the order here. Creation of the directory in daemon code is probably not the best option, esp. if it's the primary reason to *have* the privileges in the first place.